application/src/main/java/com/ericsson/bss/cassandra/ecchronos/application/DefaultNativeConnectionProvider.java

```diff
@@ -56,39 +60,38 @@ public DefaultNativeConnectionProvider(Config config, Supplier<Security.CqlSecur
                 .withPort(port)
                 .withAuthProvider(authProvider)
                 .withSslOptions(sslOptions);
-        establishTemporaryConnection(LocalNativeConnectionProvider.Builder.fromBuilder(nativeConnectionBuilder).build(),
-                host, port, nativeConfig.getTimeout());
 
-        myLocalNativeConnectionProvider = nativeConnectionBuilder
-                .build();
+        myLocalNativeConnectionProvider = establishConnection(nativeConnectionBuilder,
+                host, port, nativeConfig.getTimeout().getConnectionTimeout(TimeUnit.MILLISECONDS));
     }
 
-    private static Session establishTemporaryConnection(Cluster cluster, String host, int port, long timeout)
+    private static LocalNativeConnectionProvider establishConnection(LocalNativeConnectionProvider.Builder builder,
+                                                                     String host, int port, long timeout)
     {
         long startTime = System.currentTimeMillis();
         long endTime = startTime + timeout;
         while (endTime > System.currentTimeMillis())
         {
             try
             {
-                return cluster.connect();
+                return builder.build();
             }
             catch (NoHostAvailableException | IllegalStateException e)
             {
                 try
                 {
                     LOG.warn("Unable to connect through CQL using {}:{}, retrying", host, port);
```

Comment:
If we don't have any timeout time specified I don't think we should perform the sleep here.